### PR TITLE
fix:custom editor

### PR DESCRIPTION
**Pull request by js87zz**
### Please check if the PR fulfills these requirements
- [x] It's submitted to right branch according to our branching model
- [x] It's right issue type on title
- [x] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)
- [ ] It does not introduce a breaking change or has description for the breaking change

### Description
* add formatter option
* setColumns API
* modify custom renderer, editor for relation column

---
Thank you for your contribution to TOAST UI product. 🎉 😘 ✨

### CODE REVIEW

File: src/store/data.ts [hunk 1]
```diff
@@ -104,11 +111,15 @@ function getValidationCode(value: CellValue, validation?: Validation): Validatio
...
+  row: Row,
+  column: ColumnInfo,
+  relationMatched?: boolean,
+  relationListItems?: ListItem[]
+): CellRenderData {
   const { name, formatter, prefix, postfix, editor, validation } = column;
   let value = isRowHeader(name) ? getRowHeaderValue(row, name) : row[name];
-
-  if (related) {
+  if (relationMatched === false) {
```
Comment (seonim-ryu) [inline src/store/data.ts]: 이렇게 변경할 수 있겠네요ㅎ
`if(!relationMatched) { ... }`

Comment (js87zz) [reply src/store/data.ts]: `optional`이라..`undefined`로 넘어오는 경우에는 로직을 타면 안되서..저렇게 체크했는데..지금 좋은 로직은 아니네요. 수정하겠습니다!

File: src/formatter/listItemText.ts [hunk 2]
```diff
@@ -11,12 +11,18 @@ function getListItemText(listItems: ListItem[], value: CellValue) {
...
+  let { listItems } = column.editor!.options as ListItemOptions;
+
+  if (Array.isArray(relationListItems)) {
+    listItems = relationListItems;
+  }
   if (type === 'checkbox') {
     return String(value)
       .split(',')
       .map(getListItemText.bind(null, listItems))
+      .filter((text) => text !== '')
```
Comment (jung-han) [inline src/formatter/listItemText.ts]: `filter((text) => text.length > 0)`은 어떤가요?

Comment (js87zz) [reply src/formatter/listItemText.ts]: 수정하겠습니다!

File: src/dispatch/column.ts [hunk 3]
```diff
@@ -19,10 +19,14 @@ export function setColumns({ column, data }: Store, optColumns: OptColumn[]) {
     columnOptions,
     copyOptions,
     treeColumnOptions,
-    rowHeaders,
-    relationColumns
+    rowHeaders
   } = column.dataForColumnCreation;
 
+  const relationColumns = optColumns.reduce((acc: string[], { relations }) => {
+    acc = acc.concat(getRelationColumns(relations || []));
```
Comment (seonim-ryu) [inline src/dispatch/column.ts]: `relations` 값 구조분해할 때 초기화하고 사용하는건 어떨까요??
```js
optColumns.reduce((acc: string[], { relations = [] }) => {
  acc = acc.concat(getRelationColumns(relations);
}
```

File: src/editor/select.ts [hunk 4]
```diff
@@ -1,14 +1,20 @@
...
   public constructor(props: CellEditorProps) {
     const el = document.createElement('select');
-    const { listItems } = props.columnInfo.editor!.options as ListItemOptions;
+    const { listItems, relationListItemMap } = props.columnInfo.editor!.options as ListItemOptions;
+    let resultListItems = listItems;
 
-    listItems.forEach(({ text, value }) => {
+    if (!isEmpty(relationListItemMap) && Array.isArray(relationListItemMap![props.rowKey])) {
+      resultListItems = relationListItemMap![props.rowKey];
+    }
```
Comment (seonim-ryu) [inline src/editor/select.ts]: `checkbox`와 로직이 겹치는데 헬퍼로 분리해서 사용해도 괜찮을것 같습니다.

**Comment by seonim-ryu** [comment]
[6/17] 리뷰 완료하였습니다! 고생 많으셨어요ㅎ

File: src/editor/select.ts (hunk 4 above)
Comment (js87zz) [reply src/editor/select.ts]: 넵 수정하겠습니다!

File: src/dispatch/column.ts (hunk 3 above)
Comment (js87zz) [reply src/dispatch/column.ts]: 수정하겠습니다!

**Pull request merged**